spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/InvalidWhereBlocks.groovy

```diff
@@ -266,6 +282,560 @@ c << $spock_p_a
 '''
 
     then:
-    thrown(MissingPropertyException)
+    MissingPropertyException e = thrown()
+    e.message == 'No such property: $spock_p_a for class: apackage.ASpec'
+  }
+
+  def "table must have at least two columns"() {
+    when:
+    compiler.compileFeatureBody """
+expect:
+true
+
+where:
+a
+1
+    """
+
+    then:
+    MultipleFailuresError e = thrown()
+    e.failures*.class == [InvalidSpecCompileException] * 2
+    e.failures*.message.every { it.startsWith('where-blocks may only contain parameterizations') }
+  }
+
+  def "table with just a header is not allowed"() {
+    when:
+    compiler.compileFeatureBody """
+expect:
+true
+
+where:
+a | b
+    """
+
+    then:
+    InvalidSpecCompileException e = thrown()
+    e.message == 'Data table must have more than just the header row @ line 5, column 1.'
+  }
+
+  def "table with just a header in combination is not allowed"() {
+    when:
+    compiler.compileFeatureBody """
+expect:
+true
+
+where:
+a | b
+1 | 2
+combined:
+c | _
+    """
+
+    then:
+    InvalidSpecCompileException e = thrown()
+    e.message == 'Data table must have more than just the header row @ line 8, column 1.'
+  }
+
+  def "table with just a header is not allowed with semicolon"() {
+    when:
+    compiler.compileFeatureBody """
+expect:
+true
+
+where:
+a ; b
+    """
+
+    then:
+    InvalidSpecCompileException e = thrown()
+    e.message == 'Data table must have more than just the header row @ line 5, column 1.'
+  }
+
+  def "table with just a header in combination is not allowed with semicolon"() {
+    when:
+    compiler.compileFeatureBody """
+expect:
+true
+
+where:
+a ; b
+1 ; 2
+combined:
+c ; _
+    """
+
+    then:
+    InvalidSpecCompileException e = thrown()
+    e.message == 'Data table must have more than just the header row @ line 8, column 1.'
+  }
+
+  def "header may only contain variable names"() {
+    when:
+    compiler.compileFeatureBody """
+expect:
+true
+
+where:
+a | 1 | b
+1 | 1 | 1
+    """
+
+    then:
+    InvalidSpecCompileException e = thrown()
+    e.message.startsWith('Header of data table may only contain variable names')
```

Comment:
❔ Why is the assertion different than the next one that tests with semicolons?

Reply:
I have not the faintest idea